sklearn/tests/test_impute.py

```diff
@@ -783,21 +810,49 @@ def test_iterative_imputer_additive_matrix():
     X_test_filled = X_filled[n:]
     X_test = X_missing[n:]
 
-    imputer = IterativeImputer(n_iter=10,
-                               verbose=2,
+    imputer = IterativeImputer(max_iter=10,
+                               verbose=1,
                                random_state=rng).fit(X_train)
     X_test_est = imputer.transform(X_test)
-    assert_allclose(X_test_filled, X_test_est, atol=0.01)
+    assert_allclose(X_test_filled, X_test_est, rtol=1e-3, atol=0.01)
 
 
 def test_iterative_imputer_error_param():
     rng = np.random.RandomState(42)
     X = rng.randn(100, 2)
-    imputer = IterativeImputer(n_iter=-1)
+    imputer = IterativeImputer(max_iter=-1)
```

Comment:
You can parametrize the function and pass the parameter tol with negative number to check that the error is raised. You need to pass the params and the error message to match in the parametrization

Reply:
I'm confused here. It's just a test that tests only a single thing. Why parametrize it?

Reply:
Why not just do this
```
    imputer = IterativeImputer(max_iter=-1)
    with pytest.raises(ValueError, match='should be a positive integer'):
        imputer.fit_transform(X)
    imputer = IterativeImputer(tol=-1)
    with pytest.raises(ValueError, match='should be a non-negative float'):
        imputer.fit_transform(X)
```

Reply:
Nevermind, I'll convert as you asked.

Reply:
Your function should check that an error is raised. So you can use the parametrization to change the thing which will trigger. The core function is easier to understand then (then you need to get used to read the parametrization from pytest thought).


```python
@pytest.make.parametrize(
    "params, err_msg",
    [(-1, 'should be a positive integer'),
     (-1 'should be a non-negative float')]
)
def test_iterative_imputer_error_param("params, err_msg):
    X = np.random.randn(100, 2)
    imputer = IterativeImputer(**params)
    with pytest.raises(ValueError, match=err_msg):
        imputer.fit_transform(X)
```

Reply:
You could remove the random state there. This useless for the tests and make it easier to read the test function then.